From 7944d9779bb859da21a8f52f41987f4077de3085 Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Wed, 21 Oct 2020 17:26:03 +0530
Subject: [PATCH 0891/1921] octeontx2-pf: Remove big-endian bitfields

Big-endian support got deprecated in current and upcoming silicon's.
Hence removing the big-endian bit fields from the existing
structures.

Change-Id: Iace0227ff28a7a974b27b4494a5dd783b42c3364
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../marvell/octeontx2/nic/otx2_struct.h       | 156 ------------------
 1 file changed, 156 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_struct.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_struct.h
index fc520d30daa3..fe73d0913d87 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_struct.h
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_struct.h
@@ -99,23 +99,6 @@ struct nix_cqe_hdr_s {
 
 /* NIX CQE RX parse structure */
 struct nix_rx_parse_s {
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W0 */
-	u64 lhtype       : 4;
-	u64 lgtype       : 4;
-	u64 lftype       : 4;
-	u64 letype       : 4;
-	u64 ldtype       : 4;
-	u64 lctype       : 4;
-	u64 lbtype       : 4;
-	u64 latype       : 4;
-	u64 errcode      : 8;
-	u64 errlev       : 4;
-	u64 wqwd         : 1;
-	u64 express      : 1;
-	u64 rsvd_17      : 1;
-	u64 desc_sizem1  : 5;
-	u64 chan         : 12;
-#else
 	u64 chan         : 12;
 	u64 desc_sizem1  : 5;
 	u64 rsvd_17      : 1;
@@ -131,22 +114,6 @@ struct nix_rx_parse_s {
 	u64 lftype       : 4;
 	u64 lgtype       : 4;
 	u64 lhtype       : 4;
-#endif
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W1 */
-	u64 vtag1_tci    : 16;
-	u64 vtag0_tci    : 16;
-	u64 rsvd_95_94   : 2;
-	u64 pkind        : 6;
-	u64 vtag1_gone   : 1;
-	u64 vtag1_valid  : 1;
-	u64 vtag0_gone   : 1;
-	u64 vtag0_valid  : 1;
-	u64 l3b          : 1;
-	u64 l3m          : 1;
-	u64 l2b          : 1;
-	u64 l2m          : 1;
-	u64 pkt_lenm1    : 16;
-#else
 	u64 pkt_lenm1    : 16;
 	u64 l2m          : 1;
 	u64 l2b          : 1;
@@ -160,17 +127,6 @@ struct nix_rx_parse_s {
 	u64 rsvd_95_94   : 2;
 	u64 vtag0_tci    : 16;
 	u64 vtag1_tci    : 16;
-#endif
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W2 */
-	u64 lhflags      : 8;
-	u64 lgflags      : 8;
-	u64 lfflags      : 8;
-	u64 leflags      : 8;
-	u64 ldflags      : 8;
-	u64 lcflags      : 8;
-	u64 lbflags      : 8;
-	u64 laflags      : 8;
-#else
 	u64 laflags      : 8;
 	u64 lbflags      : 8;
 	u64 lcflags      : 8;
@@ -179,28 +135,10 @@ struct nix_rx_parse_s {
 	u64 lfflags      : 8;
 	u64 lgflags      : 8;
 	u64 lhflags      : 8;
-#endif
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W3 */
-	u64 match_id     : 16;
-	u64 pb_aura      : 20;
-	u64 wqe_aura     : 20;
-	u64 eoh_ptr      : 8;
-#else
 	u64 eoh_ptr      : 8;
 	u64 wqe_aura     : 20;
 	u64 pb_aura      : 20;
 	u64 match_id     : 16;
-#endif
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W4 */
-	u64 lhptr        : 8;
-	u64 lgptr        : 8;
-	u64 lfptr        : 8;
-	u64 leptr        : 8;
-	u64 ldptr        : 8;
-	u64 lcptr        : 8;
-	u64 lbptr        : 8;
-	u64 laptr        : 8;
-#else
 	u64 laptr        : 8;
 	u64 lbptr        : 8;
 	u64 lcptr        : 8;
@@ -209,53 +147,30 @@ struct nix_rx_parse_s {
 	u64 lfptr        : 8;
 	u64 lgptr        : 8;
 	u64 lhptr        : 8;
-#endif
-#if defined(__BIG_ENDIAN_BITFIELD)	/* W5 */
-	u64 rsvd_383_341 : 43;
-	u64 flow_key_alg : 5;
-	u64 vtag1_ptr    : 8;
-	u64 vtag0_ptr    : 8;
-#else
 	u64 vtag0_ptr    : 8;
 	u64 vtag1_ptr    : 8;
 	u64 flow_key_alg : 5;
 	u64 rsvd_383_341 : 43;
-#endif
 	u64 rsvd_447_384;		/* W6 */
 };
 
 /* NIX CQE RX scatter/gather subdescriptor structure */
 struct nix_rx_sg_s {
-#if defined(__BIG_ENDIAN_BITFIELD)	/* W0 */
-	u64 subdc      : 4;
-	u64 rsvd_59_50 : 10;
-	u64 segs       : 2;
-	u64 seg3_size  : 16;
-	u64 seg2_size  : 16;
-	u64 seg_size   : 16;
-#else
 	u64 seg_size   : 16;
 	u64 seg2_size  : 16;
 	u64 seg3_size  : 16;
 	u64 segs       : 2;
 	u64 rsvd_59_50 : 10;
 	u64 subdc      : 4;
-#endif
 	u64 seg_addr;
 	u64 seg2_addr;
 	u64 seg3_addr;
 };
 
 struct nix_send_comp_s {
-#if defined(__BIG_ENDIAN_BITFIELD)	/* W0 */
-	u64 rsvd_24_63  : 40;
-	u64 sqe_id	: 16;
-	u64 status	: 8;
-#else
 	u64 status	: 8;
 	u64 sqe_id	: 16;
 	u64 rsvd_24_63	: 40;
-#endif
 };
 
 struct nix_cqe_rx_s {
@@ -271,15 +186,6 @@ struct nix_cqe_tx_s {
 
 /* NIX SQE header structure */
 struct nix_sqe_hdr_s {
-#if defined(__BIG_ENDIAN_BITFIELD)
-	u64 sq			: 20; /* W0 */
-	u64 pnc			: 1;
-	u64 sizem1		: 3;
-	u64 aura		: 20;
-	u64 df			: 1;
-	u64 reserved_18		: 1;
-	u64 total		: 18;
-#else
 	u64 total		: 18;
 	u64 reserved_18		: 1;
 	u64 df			: 1;
@@ -287,18 +193,6 @@ struct nix_sqe_hdr_s {
 	u64 sizem1		: 3;
 	u64 pnc			: 1;
 	u64 sq			: 20;
-#endif
-#if defined(__BIG_ENDIAN_BITFIELD)    /* W1 */
-	u64 sqe_id		:16;
-	u64 il4type		:4;
-	u64 il3type		:4;
-	u64 ol4type		:4;
-	u64 ol3type		:4;
-	u64 il4ptr		:8;
-	u64 il3ptr		:8;
-	u64 ol4ptr		:8;
-	u64 ol3ptr		:8;
-#else
 	u64 ol3ptr		:8;
 	u64 ol4ptr		:8;
 	u64 il3ptr		:8;
@@ -308,27 +202,10 @@ struct nix_sqe_hdr_s {
 	u64 il3type		:4;
 	u64 il4type		:4;
 	u64 sqe_id		:16;
-
-#endif
 };
 
 /* NIX send extended header subdescriptor structure */
 struct nix_sqe_ext_s {
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W0 */
-	u64 subdc         : 4;
-	u64 mark_en       : 1;
-	u64 markform      : 7;
-	u64 markptr       : 8;
-	u64 shp_ra        : 2;
-	u64 shp_dis       : 1;
-	u64 shp_chg       : 9;
-	u64 rsvd_31_29    : 3;
-	u64 lso_format    : 5;
-	u64 lso_sb        : 8;
-	u64 tstmp         : 1;
-	u64 lso           : 1;
-	u64 lso_mps       : 14;
-#else
 	u64 lso_mps       : 14;
 	u64 lso           : 1;
 	u64 tstmp         : 1;
@@ -342,16 +219,6 @@ struct nix_sqe_ext_s {
 	u64 markform      : 7;
 	u64 mark_en       : 1;
 	u64 subdc         : 4;
-#endif
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W1 */
-	u64 rsvd_127_114  : 14;
-	u64 vlan1_ins_ena : 1;
-	u64 vlan0_ins_ena : 1;
-	u64 vlan1_ins_tci : 16;
-	u64 vlan1_ins_ptr : 8;
-	u64 vlan0_ins_tci : 16;
-	u64 vlan0_ins_ptr : 8;
-#else
 	u64 vlan0_ins_ptr : 8;
 	u64 vlan0_ins_tci : 16;
 	u64 vlan1_ins_ptr : 8;
@@ -359,22 +226,9 @@ struct nix_sqe_ext_s {
 	u64 vlan0_ins_ena : 1;
 	u64 vlan1_ins_ena : 1;
 	u64 rsvd_127_114  : 14;
-#endif
 };
 
 struct nix_sqe_sg_s {
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W0 */
-	u64 subdc	: 4;
-	u64 ld_type	: 2;
-	u64 i3		: 1;
-	u64 i2		: 1;
-	u64 i1		: 1;
-	u64 rsvd_54_50	: 5;
-	u64 segs	: 2;
-	u64 seg3_size	: 16;
-	u64 seg2_size	: 16;
-	u64 seg1_size	: 16;
-#else
 	u64 seg1_size	: 16;
 	u64 seg2_size	: 16;
 	u64 seg3_size	: 16;
@@ -385,26 +239,16 @@ struct nix_sqe_sg_s {
 	u64 i3		: 1;
 	u64 ld_type	: 2;
 	u64 subdc	: 4;
-#endif
 };
 
 /* NIX send memory subdescriptor structure */
 struct nix_sqe_mem_s {
-#if defined(__BIG_ENDIAN_BITFIELD)  /* W0 */
-	u64 subdc         : 4;
-	u64 alg           : 4;
-	u64 dsz           : 2;
-	u64 wmem          : 1;
-	u64 rsvd_52_16    : 37;
-	u64 offset        : 16;
-#else
 	u64 offset        : 16;
 	u64 rsvd_52_16    : 37;
 	u64 wmem          : 1;
 	u64 dsz           : 2;
 	u64 alg           : 4;
 	u64 subdc         : 4;
-#endif
 	u64 addr;
 };
 
-- 
2.31.1

